### ISSUE TITLE
Locally compiled assets not found on production

### ISSUE THREAD

**Issue by Anahkiasen** [jasonlewis/basset#64]
This is a Laravel 3 problem.

I have a few collections that I compiled on my local server. Everything went smoothly. I then configured my production environment, in Basset and Laravel. I deployed the repo to the production server but I'm now having a problem.
Basset is somehow stuck in `Collection->isCompiled` – the hash generated for a given assets differs from the local one which makes Basset unaware of the compiled collections in `public/assets`. Thus it displays the unconcatenated version of all assets. It's a minor annoyance bug kind of a performance hit.

Is there any way why the hash would differ between the two ? Is the base URL taken into account into it or something maybe ?

---

**UPDATE** : Apparently this has to do with the modification times of the original assets being tampered with by all FTP client. Thing is there isn't really a way to recompile the online assets is it ? I don't have console access to my shared hosting.

**Comment by jasonlewis** [comment]
@Anahkiasen Yeah it would be something to do with the modification times. Hm, you could try running the command via the web and suppressing any output. I think it L3 you can do `Command::run()`. Obviously if you have a route that does this you don't want your average joe visiting it.

Sorry about this but it's the way it works now. :/

**Comment by Anahkiasen** [comment]
Yeah I installed the [Artisan](http://bundles.laravel.com/bundle/artisan) bundle that allows me to do `http://website.com/artisan/basset::compile` so problem solved for the moment.

**Comment by jasonlewis** [comment]
Yup, I know it's not ideal but I don't plan on adding support for any web-based compiling.

**Comment by andypearson** [comment]
I've just run into exactly the same problem, also on Laravel 3.

I ended up adding a new `naming_scheme` option, which takes either `hash` the current way of handling compiled collections or `timestamp` which names the file based on when the compile command is run. Basset::show then finds the most recent timestamp and uses that as the filename for the compiled collection.

This works around the problem of the modification times of the files being different locally than when they are uploaded to a server.

Is this something you would consider as a core feature of Basset 3+4? If so, I can put together a patch...

**Comment by Anahkiasen** [comment]
I just ended up using the [Artisan](bundles.laravel.com/bundle/artisan) bundle.

**Comment by goharsahi** [comment]
And I just encountered this problem in Laravel 4. My production server is a shared-hosting server. So I copied `public` folder outside the main `laravel` folder, parallel to `laravel` folder, thanks to a very good article here http://driesvints.com/blog/laravel-4-on-a-shared-host. 

Now, locally the application was working fine, but when put online, the css and js were not at all displaying. So zipped my `assets` folder and uploaded. And then unzipped it using CPanel's file manager and that solved the issue.
